src/pcre2_chkdint.c

```diff
@@ -63,7 +63,7 @@ The INT64_OR_DOUBLE type is a 64-bit integer type when available,
 otherwise double. */
 
 BOOL
-PRIV(ckd_smul)(PCRE2_SIZE *r, int a, int b)
+PRIV(ckd_smul)(PCRE2_SIZE *r, ptrdiff_t a, ptrdiff_t b)
```

Comment:
the whole point of this function is to make sure those 2 parameters are `int`so we can check for overflow

Reply:
Ah, yeah, the hand-coded branch would need reworking.  (`__builtin_mul_overflow` should DTRT regardless.)  However, without this change, `pcre2test`'s `process_data` could in principle pass in a truncated `replen`, yielding incorrect results.  Should it instead bail if `replen > INT_MAX`?

Reply:
pcre2test is a test application, AFAIK `replen` was made a size_t just to avoid having to do a cast, and it is expected never to have a value larger than int either, so a truncation there shouldn't be possible, but feel free to poke at it and prove me wrong.

the call for ckd_smul was indeed how we are making sure that the value of the multiplication is always correct.